packages/std/src/encoding.rs

```diff
@@ -158,6 +189,84 @@ mod test {
         assert_eq!(binary.as_slice(), [0u8, 187, 61, 11, 250, 0]);
     }
 
+    #[test]
+    fn from_fixed_length_array_works() {
+        let original = &[];
+        let binary: Binary = original.into();
+        assert_eq!(binary.len(), 0);
+
+        let original = &[0u8];
+        let binary: Binary = original.into();
+        assert_eq!(binary.as_slice(), [0u8]);
+
+        let original = &[0u8, 187, 61, 11, 250, 0];
+        let binary: Binary = original.into();
+        assert_eq!(binary.as_slice(), [0u8, 187, 61, 11, 250, 0]);
+
+        let original = &[
+            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1, 1,
+        ];
+        let binary: Binary = original.into();
+        assert_eq!(
+            binary.as_slice(),
+            [
+                1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                1, 1, 1, 1,
+            ]
+        );
+    }
+
+    #[test]
+    fn from_owned_fixed_length_array_works() {
+        let original = [];
+        let binary: Binary = original.into();
+        assert_eq!(binary.len(), 0);
+
+        let original = [0u8];
+        let binary: Binary = original.into();
+        assert_eq!(binary.as_slice(), [0u8]);
+
+        let original = [0u8, 187, 61, 11, 250, 0];
+        let binary: Binary = original.into();
+        assert_eq!(binary.as_slice(), [0u8, 187, 61, 11, 250, 0]);
+
+        let original = [
+            1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+            1, 1, 1,
+        ];
+        let binary: Binary = original.into();
+        assert_eq!(
+            binary.as_slice(),
+            [
+                1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                1, 1, 1, 1,
+            ]
+        );
+    }
+
+    #[test]
+    fn from_literal_works() {
+        let a: Binary = b"".into();
+        assert_eq!(a.len(), 0);
+
+        let a: Binary = b".".into();
+        assert_eq!(a.len(), 1);
+
+        let a: Binary = b"...".into();
+        assert_eq!(a.len(), 3);
+
+        let a: Binary = b"...............................".into();
+        assert_eq!(a.len(), 31);
+
+        let a: Binary = b"................................".into();
```

Comment:
What about `b"banana".into()` ?

Reply:
There is no difference between `b"banana".into()` and `b"...............................".into()`, right? If there is we should describe that somehow.

Reply:
Just to make the secret network folks happy (the example in the issue)

Reply:
Try this: `b"🍌".into()` :rofl: